functions/Test-DbaDiskAlignment.ps1

```diff
@@ -174,24 +174,20 @@ function Test-DbaDiskAlignment {
                     if ($NoSqlCheck -eq $false) {
                         $sqldisk = $false
 
-                        foreach ($SqlInstance in $SqlInstances) {
+                        foreach ($instance in $SqlInstances) {
                             try {
-                                if ($null -ne $SqlCredential) {
-                                    $smoserver = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
-                                } else {
-                                    $smoserver = Connect-SqlInstance -SqlInstance $SqlInstance # win auth
-                                }
-                                $sql = "Select count(*) as Count from sys.master_files where physical_name like '$diskname%'"
-                                Write-Message -Level Verbose -Message "Query is: $sql" -FunctionName $FunctionName
-                                Write-Message -Level Verbose -Message "SQL Server is: $SqlInstance." -FunctionName $FunctionName
-                                $sqlcount = $smoserver.Databases['master'].ExecuteWithResults($sql).Tables[0].Count
-                                if ($sqlcount -gt 0) {
-                                    $sqldisk = $true
-                                    break
-                                }
+                                $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
                             } catch {
-                                Stop-Function -Message "Can't connect to $ComputerName ($SqlInstance)." -FunctionName $FunctionName -InnerErrorRecord $_
-                                return
+                                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+                            }
+
+                            $sql = "Select count(*) as Count from sys.master_files where physical_name like '$diskname%'"
+                            Write-Message -Level Verbose -Message "Query is: $sql" -FunctionName $FunctionName
+                            Write-Message -Level Verbose -Message "SQL Server is: $instance." -FunctionName $FunctionName
+                            $sqlcount = $server.Query($sql).Count
+                            if ($sqlcount -gt 0) {
+                                $sqldisk = $true
+                                break
```

Comment:
thanks for all of this! i wonder if we should create an issue to look for breaks and see where they're acceptable and where they are not. this is one of our older commands, so i assume the break should be a return instead.

Reply:
The break is ok here. It's inside of a test if the drive has an instance. If I found one instance, I can skip testing the other instances.